scenes/component/control/label/label_typing/label_typing.gd

```diff
@@ -0,0 +1,80 @@
+class_name LabelTyping extends Label
+
+signal typing_animation_end
+
+var _typing_delay: float
+var _loop: bool = false
+
+@onready var typing_text_tween: Node = %TypingTextTween
+
+###############
+## overrides ##
+###############
+
+
+func _ready() -> void:
+	_initialize()
+	_load_from_save_file()
+	_connect_signals()
+
+
+#############
+## helpers ##
+#############
+
+
+func _initialize() -> void:
+	pass
+
+
+func _load_from_save_file() -> void:
+	var value: float = SaveFile.effect_settings["typing"]["value"]
+	var toggle: float = SaveFile.effect_settings["typing"]["toggle"]
+	update_typing_effect(toggle, value)
+
+
+func _normalize_typing_effect(value: float) -> float:
+	return value / 4.0
+
+
+#############
+## methods ##
+#############
+
+
+func play_typing_animation(loop: bool = false) -> void:
+	_loop = loop
+	var animation_length: float = self.text.length() * _typing_delay
```

Comment:
it's maybe a bit strange that the typing gets slower when the value gets higher (compared to the shake setting that gets more when the value is higher)

Reply:
I was thinking along the lines of typing delay, then Off means no typing delay and is also equivalent to 0 on slider.

If I reverse the slider then Off corresponds to 100 on slider 🤔

Reply:
Hmm it does make more sense to see the 0 value as "infinitely fast"